### ISSUE TITLE
cli: add an Upgrade command

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#2556]
Now that `install` fails if a cluster exists, we need an `upgrade` command that fetches the controller's config to use to configure the newer installation. This command should probably NOT accept override configuration (i.e. as is accepted by install and inject), since config updates and software upgrades should, hopefully, be distinct.

**Comment by olix0r** [comment]
i have a branch that adds the basis for the command...

**Comment by alpeb** [comment]
I just wanted to clarify that according to #2564 we are accepting override config flags after all.

**Comment by olix0r** [comment]
@alpeb correct. we realized there was no way to update the configuration otherwise.
